components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/SetCookieFunctionImpl.java

```diff
@@ -68,7 +71,11 @@ public void setCookie(JsServletResponse response, String name, Object... params)
             if (sign) {
                 try {
                     String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-                    signature = Base64.encode(IdentityUtil.signWithTenantKey(value, tenantDomain));
+                    // For getCookie, setCookie functionalities tenant shouldn't use its tenanted keystore.
+                    // Hence, below code will create a keystore for this context if not exists.
+                    HTTPFunctionsServiceHolder.getInstance().getIdentityKeyStoreGenerator()
+                            .generateKeyStore(tenantDomain, KEY_STORE_CONTEXT);
+                    signature = Base64.encode(IdentityUtil.signWithTenantKey(value, tenantDomain, KEY_STORE_CONTEXT));
```

Comment:
These changes should be reflected in the CookieFunctionImpl.setCookie function as well right?

Reply:
addressed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/198/commits/f892ecc7abcb6294d96e2e49bbd896af55fe9421